src/coredump/coredump.c

```diff
@@ -1628,42 +1664,56 @@ static int forward_coredump_to_container(Context *context) {
         return 0;
 }
 
-static int gather_pid_mount_tree_fd(const Context *context) {
-        _cleanup_close_ int mntns_fd = -EBADF, root_fd = -EBADF;
+static int gather_pid_mount_tree_fd(const Context *context, int *ret_fd) {
+        _cleanup_close_ int mntns_fd = -EBADF, root_fd = -EBADF, fd = -EBADF;
         _cleanup_close_pair_ int pair[2] = EBADF_PAIR;
-        int fd = -EBADF, r;
-        pid_t child;
+        int r;
 
         assert(context);
+        assert(ret_fd);
 
         /* Don't bother preparing environment if we can't pass it to libdwfl. */
 #if !HAVE_DWFL_SET_SYSROOT
-        return -EBADF;
+        *ret_fd = -EOPNOTSUPP;
+        return 0;
 #endif
 
-        if (!arg_access_container)
-                return -EBADF;
+        if (!arg_enter_namespace) {
+                *ret_fd = -EHOSTDOWN;
+                return 0;
+        }
 
         if (socketpair(AF_UNIX, SOCK_DGRAM|SOCK_CLOEXEC, 0, pair) < 0)
                 return log_error_errno(errno, "Failed to create socket pair: %m");
 
-        r = namespace_open(context->pid,  NULL, &mntns_fd, NULL, NULL, &root_fd);
+        r = namespace_open(context->pid, NULL, &mntns_fd, NULL, NULL, &root_fd);
         if (r < 0)
                 return log_error_errno(r, "Failed to open mount namespace of crashing process: %m");
 
-        r = namespace_fork("(sd-mount-tree-ns)", "(sd-mount-tree)", NULL, 0, FORK_RESET_SIGNALS|FORK_DEATHSIG_SIGKILL, -1, mntns_fd, -1, -1, root_fd, &child);
+        r = namespace_fork("(sd-mount-tree-ns)",
+                           "(sd-mount-tree)",
+                           /* except_fds= */ NULL,
+                           /* n_except_fds= */ 0,
+                           FORK_RESET_SIGNALS|FORK_DEATHSIG_SIGKILL|FORK_LOG|FORK_WAIT,
+                           /* pidns_fd= */ -EBADF,
+                           mntns_fd,
+                           /* netns_fd= */ -EBADF,
+                           /* userns_fd= */ -EBADF,
+                           root_fd,
+                           NULL);
         if (r < 0)
                 return log_error_errno(r, "Failed to fork(): %m");
```

Comment:
now that you use FORK_LOG, drop the log message here, it would be duplicative now. just do `return r` directly here